packages/palette/src/utils/splitProps.ts

```diff
@@ -0,0 +1,26 @@
+import { styleFn } from "styled-system"
+
+/**
+ * Allows you to pass in a styled-system function and split out
+ * in a typesafe manner, the props that match and do not match
+ * the styled function.
+ */
+export const splitProps = <T>(mixin: styleFn) => {
+  const re = new RegExp(`^(${mixin.propNames.join("|")})$`)
+
+  return <U>(props: U): [T, Omit<U, keyof T>] => {
+    const leftProps = {} as T
+    const rightProps = {} as Omit<U, keyof T>
+
+    for (const key of Object.keys(props)) {
+      if (re.test(key)) {
+        leftProps[key] = props[key]
+        continue
+      }
+
+      rightProps[key] = props[key]
+    }
+
+    return [leftProps, rightProps]
+  }
+}
```

Comment:
USEFUL

Reply:
Ahh, so the generic method of the splitBoxProps. I like.